### ISSUE TITLE
Build Error - cannot unmarshal object into Go struct field DevicesResponse.data of type []appstoreconnect.Device

### ISSUE THREAD

**Issue by kndl22** [bitrise-steplib/steps-ios-auto-provision-appstoreconnect#13]
Hi,
Everytime when adding a new Test Device to Bitrise, the next build will fail with error:
`registering device Failed to register device: json: cannot unmarshal object into Go struct field DevicesResponse.data of type []appstoreconnect.Device`

The error is gone after few times of builds, but it'll surely happen next time device is added.

**Comment by chaitanyatanna-bitrise** [comment]
Hi @kndl22 
Thank you for reaching out to Bitrise. 
Can you please share the screenshot of the error that you are getting when adding a new test device?
Also, for reference here is the document that you can follow to add new test devices to your bitrise account. 

Thank you 
Chaitanya

**Comment by kndl22** [comment]
After reading [this article](https://devcenter.bitrise.io/testing/registering-a-test-device), next time I'll try adding new device while using "Safari" browser and see if it happens again.
Seems that when using Chrome the popup isn't showing (probably blocked by ad-blocker).
Will update the next time i register new  device :)

Error screenshot attached:
<img width="1188" alt="Screen Shot 2020-07-31 at 13 03 23" src="https://user-images.githubusercontent.com/6260759/89024821-697d5800-d32e-11ea-9fa9-3f1d4d623aa4.png">

**Comment by chaitanyatanna-bitrise** [comment]
Thank you @kndl22 for the update. 
Yes, please do.
Let me know if you have any questions or concerns.

Thank you
Chaitanya

**Comment by kndl22** [comment]
Hi @chaitanyatanna-bitrise, I can confirm that this issue repeats every time we add a test device manually to Bitrise (without adding it at the Apple's developer portal).

As I mentioned before, it happens only at the first build after adding the device; the next builds will pass successfuly

**Comment by chaitanyatanna-bitrise** [comment]
Hi @kndl22 
Thank you for the update. I will let the team regarding this issue.

Thank you 
Chaitanya

**Comment by ayn** [comment]
I'm also seeing this same issue, basically it fails after it registers a new device to Apple developer portal, but the device is added successfully at the portal. Rebuilding the build until all new devices are registered and the build will eventually work.

**Comment by ClaudiaJ** [comment]
I created a workflow loop to work around this when it showed up for us one late Friday night:

```yaml
  brute-force-provision:
    description: |-
      Workaround [ios-auto-provision-appstoreconnect #13](https://github.com/bitrise-steplib/steps-ios-auto-provision-appstoreconnect/issues/13).
      Attempts auto-provision with minimum setup, and on pipeline failure triggers the same workflow to be run again.
    steps:
    - activate-ssh-key@4.0:
        run_if: '{{getenv "SSH_RSA_PRIVATE_KEY" | ne ""}}'
    - git-clone@4.0: {}
    - ios-auto-provision-appstoreconnect@0.1:
        inputs:
        - verbose_log: 'yes'
        - distribution_type: "ad-hoc"
    - trigger-bitrise-workflow@0:
        run_if: ".IsBuildFailed"
        is_always_run: true
        inputs:
        - workflow_id: brute-force-provision
        - api_token: "$BITRISE_TRIGGER_TOKEN"
```

Client verbose logs are turned off in the step, so I couldn't see what the Apple API was returning that would cause the step to fail in this way:
https://github.com/bitrise-steplib/steps-ios-auto-provision-appstoreconnect/blob/c1d53fc60d082f8e32efaa0ea43231d25c1be772/main.go#L356

We don't use Bitrise test devices functionality anymore, so the register stage in this step was superfluous for us, but I couldn't disable it as-written with a config option either.

**Comment by nauticalcoder** [comment]
We just started seeing this error this morning but have not recently added any devices nor changed anything with the build process.

**Comment by kndl22** [comment]
Hey @chaitanyatanna-bitrise any news regarding the issue?

**Comment by chaitanyatanna-bitrise** [comment]
Hi,
There is still no update on the ticket from engineering.
I will update you as soon as I hear back from the team.

Thanks
Chaitanya

**Comment by adborbas** [comment]
Hi @kndl22, PR #19 solves this issue. Please open a new issue if you still experience the problem.
